src/server.ts

```diff
@@ -104,6 +111,28 @@ export async function start_server<RawConfiguration, Configuration, State>(
     }
   );
 
+  server.addHook("preHandler", (request, reply, done) => {
+    const expectedAuthHeader =
+      options.serviceTokenSecret === null
+        ? null
+        : `Bearer ${options.serviceTokenSecret}`;
+
+    const authHeader = request.headers.authorization ?? null;
+
+    if (authHeader === expectedAuthHeader) {
```

Comment:
Likewise.

Reply:
I feel like `expectedAuthHeader` is actually set to a bool above so this will always fail? Maybe I haven't had enough coffee yet this morning...

Reply:
`expectedAuthHeader` is set to either `undefined` or `Bearer <secret_token>`, depending on whether the flag was set or not.

For the auth check to succeed, the `Authorization` header should match.
This means requests with the `Authorization` header will fail if no secret was configured.